File: haskell/glue/src/Glue/Lib/Math/Logarithmic.hs
module Glue.Lib.Math.Logarithmic where

import Glue.Eval (Eval)
import Glue.Lib.Math.Logarithmic.Lg qualified as Lg
import Glue.Lib.Math.Logarithmic.Ln qualified as Ln
import Glue.Lib.Math.Logarithmic.Log qualified as Log
import Glue.Module (ModuleInfo, nativeModule)

logarithmic :: ModuleInfo Eval
logarithmic =
    nativeModule
        "ffi.math.logarithmic"
        [ ("ln", Ln.ln)
        , ("lg", Lg.lg)
        , ("log", Log.log)
        ]

File: dart/glue/lib/src/lib/math/logarithmic/logarithmic.dart
import 'package:glue/src/ir.dart';
import 'package:glue/src/module.dart';
import 'lg.dart';
import 'ln.dart';
import 'log.dart';

/// Logarithmic module - logarithm functions with various bases
/// Mirrors Haskell Glue.Lib.Math.Logarithmic exactly

/// The logarithmic module containing logarithm functions
/// Mirrors Haskell Glue.Lib.Math.Logarithmic.logarithmic exactly
final ModuleInfo logarithmic = nativeModule('ffi.math.logarithmic', [
  // Logarithm with arbitrary base (value, base)
  ('log', IrNativeFunc(log)),

  // Natural logarithm (base e)
  ('ln', IrNativeFunc(ln)),

  // Common logarithm (base 10)
  ('lg', IrNativeFunc(lg)),
]);
